src/functional/iterator/ir.py

```diff
@@ -1,11 +1,18 @@
 from typing import List, Union
 
-from eve import Node
+import eve
 from eve.traits import SymbolName, SymbolTableTrait
 from eve.type_definitions import SymbolRef
 from functional.iterator.util.sym_validation import validate_symbol_refs
 
 
+class Node(eve.Node):
+    def __str__(self):
```

Comment:
Can you add a small test for this with a couple of different nodes ?

Reply:
I added a minimal test to check the behavior of the `__str__` method (specific formatting is tested in the pretty printer).